File: guides/micronaut-scope-types/kotlin/src/main/kotlin/example/micronaut/prototype/RobotController.kt
package example.micronaut.prototype

import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

@Controller("/prototype") // <1>
class RobotController(
    private val robotMother: RobotMother, // <2>
    private val robotFather: RobotFather
) {

    @Get // <3>
    fun children(): List<String> {
        return listOf(robotMother.child().getSerialNumber(), robotFather.child().getSerialNumber())
    }
}

File: guides/micronaut-scope-types/python/src/example/micronaut/singleton/robot_controller.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.http.annotation import Get

from .robot_father import RobotFather
from .robot_mother import RobotMother

# tag::pkg[]
# end::pkg[]
# tag::imports[]
# tag::controller[]
father: Annotated[RobotFather, Inject]  # <1>
mother: Annotated[RobotMother, Inject]  # <2>
# end::controller[]
# end::imports[]


# tag::clazz[]
@Get("/singleton")  # <3>
def children() -> list[str]:
    return [
        father.child().serial_number,
        mother.child().serial_number,
    ]
# end::clazz[]
